scipy/stats/tests/test_stats.py

```diff
@@ -2469,6 +2471,14 @@ def test_onesample(self):
             assert_raises(ValueError, stats.ttest_1samp, x, 5.0,
                           nan_policy='foobar')
 
+    def test_1samp_alternative(self):
+        assert_raises(ValueError, stats.ttest_1samp, self.X1, 0, alternative="error")
+
+        t, p = stats.ttest_1samp(self.X1, 1, alternative="less")
```

Comment:
If you're going to do just one test, you should use `self.X2` instead of `self.X1` because the latter has a p-value of 1/2 with either alternative, so you could have the logic reversed and not catch it.

Reply:
Never mind, I see that you were using a hypthesized mean `mu=1`, not `mu=0`. This is fine.